nativelib/src/main/scala/scala/scalanative/native/package.scala

```diff
@@ -147,11 +148,14 @@ package object native {
     new String(bytes, charset)
   }
 
-  /** Convert a java.lang.String to a CString using given charset. */
+  /** Convert a java.lang.String to a CString using given charset.
+   *  The resulting c-style string is allocated using malloc, and
+   *  must be freed manually.
+   */
   def toCString(str: String,
                 charset: Charset = Charset.defaultCharset()): CString = {
     val bytes = str.getBytes(charset)
-    val cstr  = GC.malloc_atomic(bytes.length + 1).cast[Ptr[Byte]]
+    val cstr  = malloc(bytes.length + 1).cast[Ptr[Byte]]
```

Comment:
Actually, doing `free` is horrendously error prone, lets use memory zones instead https://github.com/scala-native/scala-native/pull/734

I'll rebase regex PR on top once the zones are merged.